mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -179,52 +179,96 @@ trait Params extends Identifiable with Serializable {
   /**
    * Sets a parameter (by name) in the embedded param map.
    */
-  private[ml] def set(param: String, value: Any): this.type = {
+  protected final def set(param: String, value: Any): this.type = {
     set(getParam(param), value)
   }
 
   /**
-   * Gets the value of a parameter in the embedded param map.
+   * Optionally returns the user-supplied value of a param.
+   */
+  final def get[T](param: Param[T]): Option[T] = {
```

Comment:
I made this change in my last commit. `set` cannot be public because `Model` also extends `Params` and some params should be immutable in models. Maybe `clear` shouldn't be public either. But for get, getDefault, and getOrDefault, I think those could be public.